[BUG] V3 Regression: `pwa-kit-dev` binary commands have implicit requirement that they execute from a directory with a `package.json`

<!--
NOTICE: GitHub is not a mechanism for receiving support under any agreement or SLA. If you require immediate assistance, please use official support channels.
-->

### Summary

_Short summary of what is going on or to provide context_.

Via an internal support conversation someone was trying to execute `npx @salesforce/pwa-kit-dev tail-logs --environment $ENV_ID --project $PROJ_ID` and getting the error `Error: Cannot find module '/Users/<user>/package.json' ` which should work from any working directory context, but fails with the error above

The fix is to avoid the `pacakge.json` filesystem read in the main pwa-kit-dev binary dir here

https://github.com/SalesforceCommerceCloud/pwa-kit/blob/a469f4be74e8cd2f58a19401a2b5e3eec2aaa098/packages/pwa-kit-dev/bin/pwa-kit-dev.js#L16

### Steps To Reproduce

1. `cd` into a directory with no `package.json`
1. run `npx @salesforce/pwa-kit-dev tail-logs --environment $ENV_ID --project $PROJ_ID`

### Expected result

1. Log tailing works

_Describe what should have happened_.

### Actual result

1. The command fails with this
<img width="1809" alt="254922182-685be93d-8a67-482f-9fd0-ad4a71b905f4" src="https://github.com/SalesforceCommerceCloud/pwa-kit/assets/47546998/ccf714dd-43ce-4ad1-bba9-87f57475ff9e">


_Describe what actually happened_.

### System Information (as applicable)

**Browser**:
**Node version**: 18
**pwa-kit version**: 3
**Desktop OS**:
**Mobile Device Info**:

### Additional information

_Feel free to attach a screenshot._

--- a/packages/pwa-kit-dev/bin/pwa-kit-dev.js
+++ b/packages/pwa-kit-dev/bin/pwa-kit-dev.js
@@ -13,7 +13,6 @@ const WebSocket = require('ws')
 const program = require('commander')
 const validator = require('validator')
 const {execSync: _execSync} = require('child_process')
-const projectPkg = require(process.cwd() + '/package.json')
 const {getConfig} = require('@salesforce/pwa-kit-runtime/utils/ssr-config')
 
 // Scripts in ./bin have never gone through babel, so we
@@ -179,20 +178,18 @@ const main = async () => {
         })
 
     const appSSRpath = p.join(process.cwd(), 'app', 'ssr.js')
-    const appSSRjs = fse.pathExistsSync(appSSRpath)
+    const appSSRexists = fse.pathExistsSync(appSSRpath)
+    const {overridesDir} = scriptUtils.getProjectPkg()?.ccExtensibility ?? {}
     const overrideSSRpath = p.join(
         process.cwd(),
-        typeof projectPkg?.ccExtensibility?.overridesDir === 'string' &&
-            !projectPkg?.ccExtensibility?.overridesDir?.startsWith(p.sep)
-            ? p.sep + projectPkg?.ccExtensibility?.overridesDir
-            : projectPkg?.ccExtensibility?.overridesDir
-            ? projectPkg?.ccExtensibility?.overridesDir
-            : '',
+        typeof overridesDir === 'string' && !overridesDir?.startsWith(p.sep)
+            ? p.sep + overridesDir
+            : overridesDir || '',
         'app',
         'ssr.js'
     )
-    const overrideSSRjs = fse.pathExistsSync(overrideSSRpath)
-    const resolvedSSRPath = appSSRjs ? appSSRpath : overrideSSRjs ? overrideSSRpath : null
+    const overrideSSRexists = fse.pathExistsSync(overrideSSRpath)
+    const resolvedSSRPath = appSSRexists ? appSSRpath : overrideSSRexists ? overrideSSRpath : null
 
     program
         .command('start')

